src/components/vm/consoles/vnc.jsx

```diff
@@ -377,17 +398,18 @@ export class VncActive extends React.Component {
         return window.location.protocol === 'https:';
     }
 
-    onDisconnected(detail) { // server disconnected
-        console.info('Connection lost: ', detail);
-        this.props.state.setConnected(false);
+    onDisconnected(event) { // server disconnected
+        console.info('Connection lost: ', event.detail);
+        this.props.state.setDisconnected(event.detail.clean);
     }
 
     onInitFailed(detail) {
         console.error('VncConsole failed to init: ', detail, this);
     }
 
     onSecurityFailure(event) {
-        console.info('Security failure:', event?.detail?.reason || "unknown reason");
+        if (event.detail?.reason)
+            this.props.state.setConnectionFailure(event.detail.reason);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2038-7e41038d-20250806-115005-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)